pxr/usd/usdShade/validators.cpp

```diff
@@ -192,13 +199,160 @@ _ShaderPropertyTypeConformance(const UsdPrim &usdPrim)
     return errors;
 }
 
+static
+UsdValidationErrorVector
+_SubsetMaterialBindFamilyName(const UsdPrim& usdPrim)
+{
+    if (!(usdPrim && usdPrim.IsInFamily<UsdGeomSubset>(
+            UsdSchemaRegistry::VersionPolicy::All))) {
+        return {};
+    }
+
+    const UsdGeomSubset subset(usdPrim);
+    if (!subset) {
+        return {};
+    }
+
+    size_t numMatBindingRels = 0u;
+
+    const std::vector<UsdProperty> matBindingProperties =
+        usdPrim.GetProperties(
+            /* predicate = */ [](const TfToken& name) {
+                return TfStringStartsWith(
+                    name, UsdShadeTokens->materialBinding);
+            }
+        );
+    for (const UsdProperty& matBindingProperty : matBindingProperties) {
+        if (matBindingProperty.Is<UsdRelationship>()) {
```

Comment:
Will be surprising to see a material:binding property on a prim which isn't a relationship. @mattyjams what do you think? I so we can clean this bit internally.

Thanks

Reply:
Rather, we should have a core usdShade validator just for this :D ... check if a prim has a material:binding property which isn't a relationship.

And clean this check here.

Reply:
Good call, thanks @tallytalwar!

I added a `MaterialBindingRelationships` validator that issues errors about non-relationship properties in 77ef087ab883ff9264236ab860b4dacb78ef87df, and a test for it in 7556b56b9c772343191abf70b2267ef614357c12.

I think we need to keep the `if (matBindingProperty.Is<UsdRelationship>()) {` here though, because this subset validator is trying to determine if there are any valid material binding relationships. If there are any properties, this particular validator doesn't care about those and wants to ignore them, but the new validator will spit errors about that.

Does that sound ok to you?

Reply:
Yeah that makes sense to me, thanks @mattyjams for jumping on the suggestion and implementing the new validator right away!